src/amber/router/pipe/router.cr

```diff
@@ -11,6 +11,7 @@ module Amber
 
       def initialize
         @routes = Radix::Tree(Route).new
+        @socket_routes = Array(NamedTuple(path: String, handler: WebSockets::Server::Handler)).new
```

Comment:
Should this also be a radix tree?  I didn't use one because I don't see there being many socket_endpoints in an application.